econml/iv/dr/_dr.py

```diff
@@ -40,14 +215,13 @@ class _BaseDRIVModelFinal:
     residual on residual regression.
     """
 
-    def __init__(self, model_final, featurizer,
-                 discrete_treatment, discrete_instrument,
-                 fit_cate_intercept, cov_clip, opt_reweighted):
+    def __init__(self, model_final, featurizer, fit_cate_intercept, cov_clip, opt_reweighted):
         self._model_final = clone(model_final, safe=False)
-        self._fit_cate_intercept = fit_cate_intercept
         self._original_featurizer = clone(featurizer, safe=False)
-        self._discrete_treatment = discrete_treatment
-        self._discrete_instrument = discrete_instrument
+        self._fit_cate_intercept = fit_cate_intercept
+        self._cov_clip = cov_clip
+        self._opt_reweighted = opt_reweighted
+
         if self._fit_cate_intercept:
```

Comment:
If we go with the DRLearner type of design, then this part also simplifies as we don't need to augment the featurizer to add an intercept.